packages/react-components/react-tree/src/components/TreeItem/renderTreeItem.tsx

```diff
@@ -1,23 +1,23 @@
 import * as React from 'react';
 import { getSlots } from '@fluentui/react-utilities';
-import type { TreeItemState, TreeItemSlots } from './TreeItem.types';
+import type { TreeItemState, TreeItemSlots, TreeItemContextValues } from './TreeItem.types';
+import { TreeItemProvider } from '../../contexts/treeItemContext';
 
 /**
  * Render the final JSX of TreeItem
  */
-export const renderTreeItem_unstable = (state: TreeItemState) => {
+export const renderTreeItem_unstable = (state: TreeItemState, contextValues: TreeItemContextValues) => {
   const { slots, slotProps } = getSlots<TreeItemSlots>(state);
 
   return (
-    <slots.groupper {...slotProps.groupper}>
-      <slots.root {...slotProps.root}>
-        {slots.expandIcon && <slots.expandIcon {...slotProps.expandIcon} />}
-        {slots.iconBefore && <slots.iconBefore {...slotProps.iconBefore} />}
-        {slotProps.root.children}
-        {slots.iconAfter && <slots.iconAfter {...slotProps.iconAfter} />}
-        {slots.badges && <slots.badges {...slotProps.badges} />}
-        {slots.actions && <slots.actions {...slotProps.actions} />}
-      </slots.root>
-    </slots.groupper>
+    <TreeItemProvider value={contextValues.treeItem}>
+      <slots.groupper {...slotProps.groupper}>
+        <slots.root {...slotProps.root}>
+          {slots.expandIcon && <slots.expandIcon {...slotProps.expandIcon} />}
+          {slotProps.root.children}
+          {slots.actions && <slots.actions {...slotProps.actions} />}
```

Comment:
why are actions still in the base tree item and not the layout? IMO it's not as 'importnt' to the tree item as expand icon

Reply:
Since it's a common logic between `TreeItemLayout` and `TreeItemPersonaLayout` I've decided to keep it as part of the base `TreeItem` component to reuse this logic between layouts.

Reply:
Might consider to move it to layouts in the future 🤔